src/EventListener/AssetsInstallCommandListener.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\EventListener;
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Event\ConsoleTerminateEvent;
+use Symfony\Component\Console\Exception\InvalidArgumentException;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Filesystem\Exception\IOException;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+/**
+ * This listener extends `assets:install` command when SonataCoreBundle will be not register. Files from `Resources/private/SonataCoreBundleAssets`
+ * will be copy with the same result like SonataCoreBundle is register.
+ *
+ * This class should be remove when support for Bootstrap 3 will be ended or assets system will be remove in favor for encore webpack.
+ */
+final class AssetsInstallCommandListener
+{
+    public const METHOD_COPY = 'copy';
+    public const METHOD_ABSOLUTE_SYMLINK = 'absolute symlink';
+    public const METHOD_RELATIVE_SYMLINK = 'relative symlink';
+
+    protected static $defaultName = 'assets:install';
+
+    private $filesystem;
+    private $projectDir;
+
+    public function __construct(Filesystem $filesystem, ?string $projectDir = null)
+    {
+        if (null === $projectDir) {
+            @trigger_error(sprintf('Not passing the project directory to the constructor of %s is deprecated since Symfony 4.3 and will not be supported in 5.0.', __CLASS__), E_USER_DEPRECATED);
+        }
+
+        $this->filesystem = $filesystem;
+        $this->projectDir = $projectDir;
+    }
+
+    public function copySonataCoreBundleAssets(ConsoleTerminateEvent $event)
+    {
+        $command = $event->getCommand();
+        $application = $command->getApplication();
+
+        try {
+            $coreBundle = $application->getKernel()->getBundle('SonataCoreBundle');
+        } catch (\Exception $e) {
```

Comment:
Shouldn't we catch `\Throwable`?

Reply:
Here mybe, but it work.

Reply:
Here mybe, but it work.

Reply:
InvalidArgumentException -> LogicException -> Exception
```php
    /**
     * Returns a bundle.
     *
     * @param string $name Bundle name
     *
     * @return BundleInterface A BundleInterface instance
     *
     * @throws \InvalidArgumentException when the bundle is not enabled
     */
    public function getBundle($name);
```